src/lib/offline-api/index.ts

```diff
@@ -341,6 +363,46 @@ class OfflineAPI {
 
     // Mutate version bump
     entry.version = entry.version + 1
+    entry.isPublished = true
+
+    return entry
+  }
+
+  async unpublishEntry (id: string): Promise<Entry> {
+    this.assertRecording()
+
+    const hasEntry = this.entries.some((entry) => entry.id === id)
+
+    if (!hasEntry) {
+      throw new Error(`Cannot unpublish Entry ${id} because it does not exist`)
+    }
+
+    const entry = this.entries.find((entry) => entry.id === id)
+
+    // Store clone as a request
+    this.currentRequestsRecorded.push(unpublishEntryRequest(entry.clone()))
+
+    // Mutate version bump
+    entry.version = entry.version + 1
+    entry.isPublished = false
```

Comment:
I think in addition to that we need to set `entry.sys.publishedVersion` to `undefined`.